sale_order_route/views/sale_order_view.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+         <record model="ir.ui.view" id="view_order_form">
+            <field name="name">sale.order.route.form</field>
+            <field name="model">sale.order</field>
+            <field name="inherit_id" ref="sale_stock.view_order_form_inherit" />
+            <field name="arch" type="xml">
+                <field name="warehouse_id" position="after">
+                    <field name="route_id" groups="sale_stock.group_route_so_lines"/>
+                </field>
+                <field name="order_line" position="attributes">
+                    <attribute name="context">{'default_route_id': route_id}</attribute>
```

Comment:
Maybe you should use "web_context_tunnel" module in order to avoid problems with "replacing" the context.

@pedrobaeza am I wrong?